r1cs/gadgets/std/src/groups/curves/short_weierstrass/short_weierstrass_jacobian.rs

```diff
@@ -610,196 +798,52 @@ where
-        //     [bits[n], bits[n-1],..., bits[2], 1] * T.
-        //
-        // This output is equal to 0 (and hence causes an exception in the
-        // last add when processing bits[2]) iff
-        //
-        //     [bits[n], bits[n-1],....,bits[2], 1] = p,
-        //
-        // or equivalently [bit[n],...,bits[2],bits[1]] = {p or p-1}.
-        // This corresponds to
-        //     scalar + p  = 2 * {p or p-1} + bits[0]
-        // or  scalar being from {p-2, p-1, p, p + 1}.
-        // A more detailed exploration shows that this set is the complete set
-        // of exceptions.
-        double_and_add_step(cs.ns(|| "bit 2"), &bits[2], &mut acc, &t, true)?;
-
-        // Why the step processing bit[1] needs to be treated with a
-        // secure add:
-        // After processing bit[1] `acc` is equal to
-        //
-        //                 n+1 bits
-        //     /---------------------------------------\
-        //     [bits[n], bits[n-1],..., bits[2], bits[1], 1] * T
-        //
-        // which is 0 iff
-        //
-        //     [bits[n], bits[n-1],....,bits[2],bits[1], 1] = p,
-        //
-        // or equivalently [bit[n],...,bits[2],bits[1],bits[0]] = {p or p-1}.
-        // These cases corresponds to
-        //     scalar + p  = {p or p-1}
-        // or a scalar from  {0 , -1}. The latter cannot be achieved by an
-        // unsigned integer representation. The case scalar = 0 is not
-        // covered by the secure add from the step of bits[2].
-        double_and_add_step(cs.ns(|| "bit 1"), &bits[1], &mut acc, &t, true)?;
-
-        // The final bit is taken into account by correcting the `1` in
-        //
-        //     [bits[n], bits[n-1],....,bits[2],bits[1], 1] * T,
-        //
-        // via substracting T and a subsequent conditional choice
-        //
-        //      return (k[0] = 0) ? (Acc - T) : Acc.
-        //
-        // We hit exceptions in this sub if and only if
-        //
-        //   [bits[n], bits[n-1],....,bits[2],bits[1], 1]*T = +/- T,
-        // or
-        //  [bits[n], bits[n-1],....,bits[2],bits[1], 1] = +/- 1  mod p.
-        // Hence
-        //
-        //  [bits[n], bits[n-1],....,bits[2],bits[1], 0] = {-2,0} mod p,
-        //
-        // and [bits[n],...,bits[1],bits[0]] is contained in {-2,-1,0,1} mod p.
-        // As this case is already covered by the secure add of step bit[2],
-        // we may use add_unsafe() here.
-        let neg_t = t.negate(cs.ns(|| "neg T"))?;
-        let acc_minus_t = acc.add_unsafe(cs.ns(|| "Acc - T"), &neg_t)?;
-
-        let result = Self::conditionally_select(
-            cs.ns(|| "select acc or acc - T"),
-            &bits[0],
-            &acc,
-            &acc_minus_t,
+        let non_zero_curve_point = <Jacobian<P> as Curve>::prime_subgroup_generator();
+        // check that the generated point can be employed in `mul_bits_internal`
+        assert!(!non_zero_curve_point.is_zero() && non_zero_curve_point.is_valid());
```

Comment:
This check is not really needed since non_zero_curve_point is the generator (and we assume it's fine as we have specific tests for it)

Reply:
Ok